pkg/client/queue/priority_factor.go

```diff
@@ -0,0 +1,48 @@
+package queue
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"reflect"
+)
+
+type PriorityFactor float64
```

Comment:
It seems to me that this adds unnecessary complexity. Why do we need to introduce a custom type for a priority factor instead of using `float64`?

Reply:
Float64 means that any float64 value is a valid priority factor value, which isn't the case. Priority factor has to be greater than or equal to 1.0. ([reference](https://github.com/G-Research/armada/blob/2d63f9dd154f3462c1f426b28954fdf5619840cf/internal/armada/server/submit.go#L126-L131))

Reply:
Sure, but that could be checked when validating the queue struct itself. Unless I'm mistaken, creating a new queue with `Queue{}` would result in a queue for which the priority factor is `0` (the default value for a `float64`), so we need to validate the queue struct regardless of if the priority factor is a `float64` or a custom type.

Reply:
You can always instantiate invalid data structure even in standard library, Go language doesn't prevent you from improper usage.

What this does ensure, is that float64 value cannot be assigned to PriorityFactor as they are not the same type. To Generate PriorityFactor from float64 `NewPriorityFactor` should be used instead.